packages/gatsby-image/src/index.js

```diff
@@ -176,7 +176,12 @@ function generateImageSources(imageVariants) {
           sizes={sizes}
         />
       )}
-      <source media={media} srcSet={srcSet} sizes={sizes} />
+      <source
+        type={imageMediaType}
```

Comment:
`imageVariants` does not refer to the individual sizes you'd get with `srcSet`, that's why we're mapping `imageVariants`, this is an array of image objects, it's for supporting Art Direction where image variations (including completely different images) can be provided. That's where the `media` attribute assists with `<source>` selection.

In usage without Art Direction, which is most common, this would only be a single `imageVariant` item, with the intended image format and optionally additional WEBP source.

Thus this would introduce a bug, since a user might choose to provide a different image format for desktop and mobile breakpoints. You could avoid this by iterating through `imageVariants` prior to append your `imageMediaType` property to each variant object. Best place for that is right before the `return` line.